docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,188 @@
+the real and imaginary values.
+
+We support many functions for Complex Tensors eg. :func:`torch.svd`, :func:`torch.qr` etc. Operations on complex tensors are likely to be
+faster than operations on float tensors mimicking them. Operations involving complex numbers in PyTorch are
+optimized to use vectorized assembly instructions and specialized kernels (e.g. LAPACK, CuBlas). Thus using
+functions for complex tensors will provide performance benefits as opposed to users defining their own functions.
+
+.. warning ::
+     Complex Tensors is a beta feature and subject to change.
+
+Creating Complex Tensors
+----------------------
+
+We support two complex dtypes: `torch.cfloat` and `torch.cdouble`
+
+::
+
+     >>> x = torch.randn(2,2, dtype=torch.cfloat)
+     >>> x
+     tensor([[-0.4621-0.0303j, -0.2438-0.5874j],
+          [ 0.7706+0.1421j,  1.2110+0.1918j]])
+
+.. note::
+
+     The default dtype for complex tensors is determined by the default floating point dtype.
+     If the default floating point dtype is torch.float64 then complex numbers are inferred to
+     have a dtype of torch.complex128, otherwise they are assumed to have a dtype of torch.complex64.
+
+The following factory functions can be used to create complex tensors:
+
+- :func:`torch.tensor`
+- :func:`torch.as_tensor`
+- :func:`torch.empty`
+- :func:`torch.rand`
+- :func:`torch.randn`
+- :func:`torch.as_strided`
+- :func:`torch.from_numpy`
+- :func:`torch.zeros`
+- :func:`torch.ones`
+- :func:`torch.full`
+- :func:`torch.eye`
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real` view functions:
+
+::
+
+     >>> x = torch.randn(3, 2)
+     >>> x
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+     >>> y = torch.view_as_complex(x)
+     >>> y
+     tensor([ 0.6125-0.1681j, -0.3773+1.3487j, -0.0861-0.7981j])
+     >>> torch.view_as_real(y)
+     tensor([[ 0.6125, -0.1681],
+          [-0.3773,  1.3487],
+          [-0.0861, -0.7981]])
+
+Accessing real and imag
+-----------------------
+
+The real and imaginary values of a complex tensor can be accessed using the :attr:`real` and
+:attr:`imag` views.
+
+::
+
+     >>> y.real
+     tensor([ 0.6125, -0.3773, -0.0861])
+     >>> y.imag
+     tensor([-0.1681,  1.3487, -0.7981])
+
+Angle and abs
+-------------
+
+The angle and absolute values of a complex tensor can be accesses using :func:`torch.angle` and
+`torch.abs`.
+
+::
+
+     >>> x1=torch.tensor([3j, 4+4j])
+     >>> x1.abs()
+     tensor([3.0000, 5.6569])
+     >>> x1.angle()
+     tensor([1.5708, 0.7854])
+
+Linear Algebra
+--------------
+
+Currently, there is very minimal linear algebraic operation support for complex tensors.
```

Comment:
"linear algebraic" isn't very idiomatic here. I'd probably personally write "there is very minimal support for linear algebra operations"